src/fi_tostr.c

```diff
@@ -492,6 +492,38 @@ static void fi_tostr_version(char *buf)
 	strcatf(buf, VERSION);
 }
 
+static void fi_tostr_eq_event(char *buf, int type)
+{
+	switch (type) {
+	CASEENUMSTR(FI_NOTIFY);
+	CASEENUMSTR(FI_CONNREQ);
+	CASEENUMSTR(FI_CONNECTED);
+	CASEENUMSTR(FI_SHUTDOWN);
+	CASEENUMSTR(FI_MR_COMPLETE);
+	CASEENUMSTR(FI_AV_COMPLETE);
+	default:
+		strcatf(buf, "Unknown");
+		break;
+	}
+}
+
+static void fi_tostr_cq_event_flags(char *buf, uint64_t flags)
+{
+	IFFLAGSTR(flags, FI_SEND);
+	IFFLAGSTR(flags, FI_RECV);
+	IFFLAGSTR(flags, FI_RMA);
+	IFFLAGSTR(flags, FI_ATOMIC);
+	IFFLAGSTR(flags, FI_MSG);
+	IFFLAGSTR(flags, FI_TAGGED);
+	IFFLAGSTR(flags, FI_READ);
+	IFFLAGSTR(flags, FI_WRITE);
+	IFFLAGSTR(flags, FI_REMOTE_READ);
+	IFFLAGSTR(flags, FI_REMOTE_WRITE);
+	IFFLAGSTR(flags, FI_REMOTE_CQ_DATA);
+	IFFLAGSTR(flags, FI_MULTI_RECV);
+	fi_remove_comma(buf);
+}
```

Comment:
I wonder if we shouldn't enhance fi_tostr_flags to handle more of the generic flags.  The comment in fabric.h regarding flag name space isn't correct anymore, with the CQ flags now defined.  Maybe operation specific flags should be bumped from bits 0-10 to bits 50-59.  There are only a couple of those flags defined anyway.  Or we need to redefine the values of FI_MSG/RMA/TAGGED/ATOMIC.